test/e2e/build-sample-external-plugin.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+# TODO: Dynamically set exteranl plugin destination based on OS platform
+EXTERNAL_PLUGIN_DESTINATION_PREFIX="${HOME}/Library/Application Support/kubebuilder/plugins"
```

Comment:
This is making me wonder if it would be worth implementing the functionality to configure where kubebuilder should search for external plugins sooner rather than later. I'm not a huge fan of having e2e tests making modifications to a HOME directory.

Reply:
I completely agree with your assessment. 

Modifying the HOME directory during e2e testing is indeed not an ideal scenario, and it may lead to unforeseen issues or conflicts.

I think allowing users to configure the path for searching external plugins would significantly increase the flexibility and control over the system. 

Wondering if it's possible to read a configuration file or an environment variable provided by the user, to specify the desired path for the plugin search? Hope in this way we could avoid making assumptions about the user's directory structure or enforcing unnecessary constraints.

Reply:
One approach maybe:
Introduce an env var `KB_PLUGIN_PATH` when executing the kb command to tell the path of the external plugin.
```
KB_PLUGIN_PATH="/opt/app/kubebuilder/plugin" \
  kubebuilder create api --plugins=myexternalplugin/v1
```
If such env does not exist, fall back to the default (currently used) path.